{"definition":{"tags":["aws","az","kubernetes"],"title":"Losing an AZ should not impact availability","method":[{"name":"Inject mild load into our application","type":"action","provider":{"path":"ddosify","type":"process","arguments":"-d 330 -n 150 -o stdout-json -t http://k8s-default-lbdemo-b42659e47c-1929532341.eu-central-1.elb.amazonaws.com/consumer/data"},"background":true},{"name":"Simulate AZ Failure for ALB","type":"action","pauses":{"after":300},"provider":{"func":"fail_az","type":"python","module":"azchaosaws.elbv2.actions","arguments":{"az":"eu-central-1b","tags":[{"Key":"ingress.k8s.aws/stack","Value":"default/lb-demo"}],"dry_run":false}}}],"secrets":{"slack":{"token":{"key":"SLACK_TOKEN","type":"env"}}},"controls":[{"name":"reliably","provider":{"type":"python","module":"chaosreliably.controls.experiment","arguments":{"exp_id":"4e6f6a1f-ebac-41c0-be7b-ac79803d896a","org_id":"8f67bb6a-4944-40af-80b0-921023467cdc"}}},{"name":"slack","provider":{"type":"python","module":"chaosslack.control","secrets":["slack"],"arguments":{"channel":"chaosdemo"}}},{"name":"opentracing","provider":{"type":"python","module":"chaostracing.control"}},{"name":"reliably","provider":{"type":"python","module":"chaosreliably.controls.experiment","arguments":{"exp_id":"5653c16d-1e17-44b4-9f12-2fea9b246749","org_id":"8f67bb6a-4944-40af-80b0-921023467cdc"}}}],"rollbacks":[{"name":"Recover AZ Failure for ALB","type":"action","provider":{"func":"recover_az","type":"python","module":"azchaosaws.elbv2.actions"}}],"description":"Remove all subnets of a zone from the load balancer so that zone cannot be accessed anymore from the internet","configuration":{"aws_region":"eu-central-1","tracing_provider":"opentelemetry","tracing_opentelemetry_exporter":"oltp-http","tracing_opentelemetry_baggage_prefix":"b3","tracing_opentelemetry_collector_endpoint":"https://api.honeycomb.io/v1/traces"},"contributions":{"latency":"low","security":"none","availability":"high"},"steady-state-hypothesis":{"title":"Users should always see the service available","probes":[{"name":"get-lb-unavailable-service-error-count-during-the-last-5m","type":"action","provider":{"func":"get_metric_statistics","type":"python","module":"chaosaws.cloudwatch.probes","arguments":{"duration":300,"namespace":"AWS/ApplicationELB","statistic":"Sum","metric_name":"HTTPCode_ELB_503_Count","dimension_name":"LoadBalancer","dimension_value":"app/k8s-default-lbdemo-b42659e47c/bcdbe82dbd3b2dea"}},"tolerance":0}]}},"id":"5653c16d-1e17-44b4-9f12-2fea9b246749","org_id":"8f67bb6a-4944-40af-80b0-921023467cdc","created_date":"2022-09-22T09:37:09.491857+00:00"}